### ISSUE TITLE
Accessibility: Non-Descriptive form field labels

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51793]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

This input has a corresponding label, however the nearby descriptive text that provides additional context is not associated with the input, so screen reader users do not have access to the additional descriptive information. 

![T001_SR_InfoNotAnnounced](https://github.com/user-attachments/assets/dc9e19b4-d86c-4a37-93ac-341996ed636f)

### Component or Page: 

**Page: Checkout Process (Block-Based)**

### Name affected element:

Email Address edit field when logged out with additional message: 'You are currently checking out as a guest.'

### Relevant code snippets:

`<div id="contact" class="wc-block-components-address-form"><div class="wc-block-components-text-input wc-block-components-address-form__email is-active"><input type="text" id="email" autocapitalize="none" autocomplete="email" aria-label="Email address" required="" aria-invalid="false" title="" value="test23@gmail.com"><label for="email">Email address</label></div><p class="wc-block-checkout__guest-checkout-notice">You are currently checking out as a guest.</p></div>`

### Recommended fix:

Use the aria-describedby attribute to connect the additional descriptive information to the input.

**Comment by Manussakis** [comment]
I have started working on this issue.
